Helpers/CommonMenuItemPartMigrator.cs

```diff
@@ -0,0 +1,128 @@
+                .ListAsync();
+
+            foreach (var contentItem in contentItems)
+            {
+                var menuItemsListPart = contentItem.As<MenuItemsListPart>();
+
+                foreach (var linkMenuItem in menuItemsListPart.MenuItems)
+                {
+                    var linkDestinationPart = new LinkDestinationPart();
+
+                    try
+                    {
+                        if (linkMenuItem.As<CommonMenuItem>().ExternalUrl.Text != null)
+                        {
+                            linkDestinationPart = new LinkDestinationPart { DestinationUrl = linkMenuItem.As<CommonMenuItem>().ExternalUrl }; ;
+                        }
+                        else
+                        {
+                            var host = _httpContextAccessor.HttpContext.Request.Host.Value;
+                            var path = _httpContextAccessor.HttpContext.Request.Path.Value != null ? _httpContextAccessor.HttpContext.Request.Path.Value.Replace("admin","") : "/" ;
+
+                            linkDestinationPart = new LinkDestinationPart
+                            {
+                                DestinationUrl = new TextField { Text = $"//{host}{path}{await ContentItemIdToUrl(linkMenuItem)}" }
+                            };
+                        }
+
+                        var linkBehaviourPart = new LinkBehaviourPart
+                        {
+                            ClickEvent = linkMenuItem.As<CommonMenuItem>().OnClick,
+                            OpenIn = new TextField { Text = linkMenuItem.As<CommonMenuItem>().OpenNewTab.Value ? "_blank" : "_self" }
+                        };
+
+                        linkMenuItem.Apply(nameof(LinkDestinationPart), linkDestinationPart);
+                        linkMenuItem.Apply(nameof(LinkBehaviourPart), linkBehaviourPart);
+                        ContentExtensions.Apply(linkMenuItem, linkMenuItem);
+                    }
+                    catch (Exception ex)
+                    {
+                        _logger.LogError(ex, $"Failed to update menu item: {linkMenuItem.ContentType} ID : {linkMenuItem.ContentItemId}");
+                        continue;
+                    }
+                }
+
+                contentItem.Apply(nameof(MenuItemsListPart), menuItemsListPart);
+                await _contentManager.UpdateAsync(contentItem);
+                await _contentManager.PublishAsync(contentItem);
+            };
+
+        }
+
+        public async Task<string> ContentItemIdToUrl(ContentItem linkMenuItem)
+        {
+            try
+            {
+                var contentItem = await _session.Query<ContentItem>()
+                    .With<ContentItemIndex>()
+                        .Where(x => x.ContentItemId == linkMenuItem.As<CommonMenuItem>().LinkTo.ContentItemIds[0])
+                    .FirstOrDefaultAsync();
+
+                return contentItem.As<AutoroutePart>().Path;

```

Comment:
Potential null reference exception here in multiple ways.

What happens when there is no `contentItem`? What happens when the content item doesn't have an `AutoroutePart`? These scenarios must be catered for.